tools/merger.py

```diff
@@ -153,16 +143,15 @@ def get_slab_halo(filenames, minified):
     N_halos_slabs = np.zeros(n_slabs, dtype=int)
 
     # extract number of halos in each slab
-    for i in range(len(filenames)):
-        fn = filenames[i]
+    for i,fn in enumerate(filenames):
         print("File number %i of %i" % (i, len(filenames) - 1))
         f = asdf.open(fn, lazy_load=True, copy_arrays=True)
-        N_halos = f.tree["data"]["HaloIndex"].shape[0]
+        N_halos = len(f["data"]["HaloIndex"])
         N_halos_slabs[i] = N_halos
 
     # sort in slab order
     i_sort = np.argsort(slabs)
     slabs = slabs[i_sort]
     N_halos_slabs = N_halos_slabs[i_sort]
 
-    return N_halos_slabs, slabs
+    return dict(zip(slabs, N_halos_slabs))
```

Comment:
This is mostly a matter of taste, but when you have two parallel arrays (i.e. same length and each element in one array has a logical correspondence to the element of the same index in the other array), consider using a dict.  Then it's one variable instead of two, and that "parallel array" idea you have in your head is explicit in the variable structure.